src/InvestmentManager.sol

```diff
@@ -567,8 +532,11 @@ contract InvestmentManager is Auth {
         auth
         returns (uint256 currencyAmount)
     {
-        currencyAmount = previewRedeem(liquidityPool, owner, trancheTokenAmount);
-        _processRedeem(investments[liquidityPool][owner], currencyAmount.toUint128(), liquidityPool, receiver, owner);
+        InvestmentState storage state = investments[liquidityPool][owner];
+        uint128 currencyAmount_ =
+            _calculateCurrencyAmount(trancheTokenAmount.toUint128(), liquidityPool, state.depositPrice);
```

Comment:
This should be redeemPrice right?
```suggestion
            _calculateCurrencyAmount(trancheTokenAmount.toUint128(), liquidityPool, state.redeemPrice);
```

Reply:
Ah, good catch. The fact that no tests are catching this is not great. Something to look into afterwards